command/tls_cert_create.go

```diff
@@ -0,0 +1,289 @@
+    Defaults to dc1.
+
+  -domain
+    Provide the domain. Matters only for -server certificates.
+  
+  -key
+    Provide path to the key. Defaults to #DOMAIN#-agent-ca-key.pem.
+
+  -node
+    When generating a server cert and this is set an additional dns name is 
+    included of the form <node>.server.<datacenter>.<domain>.
+
+  - server
+    Generate server certificate.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *TLSCertCreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-additional-dnsname":   complete.PredictAnything,
+			"-additional-ipaddress": complete.PredictAnything,
+			"-ca":                   complete.PredictAnything,
+			"-cli":                  complete.PredictNothing,
+			"-client":               complete.PredictNothing,
+			"-days":                 complete.PredictAnything,
+			"-dc":                   complete.PredictAnything,
+			"-domain":               complete.PredictAnything,
+			"-key":                  complete.PredictAnything,
+			"-node":                 complete.PredictAnything,
+			"-server":               complete.PredictNothing,
+		})
+}
+
+func (c *TLSCertCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *TLSCertCreateCommand) Synopsis() string {
+	return "Create a new certificate"
+}
+
+func (c *TLSCertCreateCommand) Name() string { return "tls cert create" }
+
+func (c *TLSCertCreateCommand) Run(args []string) int {
+
+	var (
+		dnsnames    flags.StringFlag
+		ipaddresses flags.StringFlag
+		ca          string
+		cli         bool
+		client      bool
+		key         string
+		days        int
+		dc          string
+		domain      string
+		node        string
+		server      bool
+	)
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.Var(&dnsnames, "additional-dnsname", "")
+	flags.Var(&ipaddresses, "additional-ipaddress", "")
+	flags.StringVar(&ca, "ca", "#DOMAIN#-agent-ca.pem", "")
+	flags.BoolVar(&cli, "cli", false, "")
+	flags.BoolVar(&client, "client", false, "")
+	flags.StringVar(&key, "key", "#DOMAIN#-agent-ca-key.pem", "")
+	flags.IntVar(&days, "days", 365, "")
+	flags.StringVar(&dc, "dc", "dc1", "")
+	flags.StringVar(&domain, "domain", "nomad", "")
+	flags.StringVar(&node, "node", "", "")
+	flags.BoolVar(&server, "server", false, "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments
+	args = flags.Args()
+	if l := len(args); l < 0 || l > 1 {
+		c.Ui.Error("This command takes up to one argument")
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+	if ca == "" {
+		c.Ui.Error("Please provide the ca")
+		return 1
+	}
+	if key == "" {
+		c.Ui.Error("Please provide the key")
+		return 1
+	}
+	if !((server && !client && !cli) ||
+		(!server && client && !cli) ||
+		(!server && !client && cli)) {
```

Comment:
This could be simplified I think? 
```suggestion
	if (!server || client || cli) && (server || !client || cli) && (!server || client || !cli) {
```
Or is that simpler? I'm not sure. Just a suggestion.

Reply:
The tests fail if I change the if statement to what you suggest. I need to understand your suggestion a bit better.

Reply:
I have decided to leave this as is for now.